ini/stm32h7.ini

```diff
@@ -59,3 +59,55 @@ build_flags                 = ${stm32_variant.build_flags}
                               -DD_CACHE_DISABLED
 upload_protocol             = cmsis-dap
 debug_tool                  = cmsis-dap
+
+#
+# BigTreeTech SKR V3.0 / SKR V3.0 EZ / Octopus V1.x / Octopus Pro V1.0 / Octopus Max EZ V1.0 (STM32H723VET6 ARM Cortex-M7)
+#
+[env:STM32H723Vx_btt]
+extends                     = stm32_variant
+platform                    = ststm32@~14.1.0
+platform_packages           = framework-arduinoststm32@https://github.com/stm32duino/Arduino_Core_STM32/archive/main.zip
+board                       = marlin_STM32H723Vx
+board_build.offset          = 0x20000
+board_upload.offset_address = 0x08020000
+build_flags                 = ${stm32_variant.build_flags}
+                              -DPIN_SERIAL1_RX=PA_10 -DPIN_SERIAL1_TX=PA_9
+                              -DPIN_SERIAL2_RX=PD_6 -DPIN_SERIAL2_TX=PD_5
+                              -DPIN_SERIAL3_RX=PD_9 -DPIN_SERIAL3_TX=PD_8
+                              -DPIN_SERIAL4_RX=PA_1 -DPIN_SERIAL4_TX=PA_0
+                              -DPIN_SERIAL7_RX=PE_7 -DPIN_SERIAL7_TX=PE_8
+                              -DSERIAL_RX_BUFFER_SIZE=1024 -DSERIAL_TX_BUFFER_SIZE=1024
+                              -DTIMER_SERVO=TIM5 -DTIMER_TONE=TIM2
+                              -DSTEP_TIMER_IRQ_PRIO=0
+                              -DRCC_PERIPHCLK_I2C35=RCC_PERIPHCLK_I2C5
+                              -DUSE_USB_HS -DUSE_USB_HS_IN_FS
+                              -DD_CACHE_DISABLED
+                              -UI2C5_BASE
+upload_protocol             = cmsis-dap
+debug_tool                  = cmsis-dap
+
+#
+# BigTreeTech Octopus V1.x / Octopus Pro V1.0 / Octopus Max EZ V1.0 (STM32H723ZET6 ARM Cortex-M7)
+#
+[env:STM32H723Zx_btt]
+extends                     = stm32_variant
+platform                    = ststm32@~14.1.0
+platform_packages           = framework-arduinoststm32@https://github.com/stm32duino/Arduino_Core_STM32/archive/main.zip
+board                       = marlin_STM32H723Zx
+board_build.offset          = 0x20000
+board_upload.offset_address = 0x08020000
+build_flags                 = ${stm32_variant.build_flags}
+                              -DPIN_SERIAL1_RX=PA_10 -DPIN_SERIAL1_TX=PA_9
+                              -DPIN_SERIAL2_RX=PD_6 -DPIN_SERIAL2_TX=PD_5
+                              -DPIN_SERIAL3_RX=PD_9 -DPIN_SERIAL3_TX=PD_8
+                              -DPIN_SERIAL4_RX=PA_1 -DPIN_SERIAL4_TX=PA_0
+                              -DPIN_SERIAL7_RX=PE_7 -DPIN_SERIAL7_TX=PE_8
+                              -DSERIAL_RX_BUFFER_SIZE=1024 -DSERIAL_TX_BUFFER_SIZE=1024
+                              -DTIMER_SERVO=TIM5 -DTIMER_TONE=TIM2
+                              -DSTEP_TIMER_IRQ_PRIO=0
+                              -DRCC_PERIPHCLK_I2C35=RCC_PERIPHCLK_I2C5
+                              -DUSE_USB_HS -DUSE_USB_HS_IN_FS
+                              -DD_CACHE_DISABLED
+                              -UI2C5_BASE
```

Comment:
It looks like we could define many of these, like `PIN_SERIAL1_RX`, in the `variant_*.h` file(s) (with `#ifndef`), and then only include the overrides in the env where they differ from typical defaults. Do board makers have wide discretion about these pins, or are these the serial ports that would always be defined based on the MCU?

Reply:
@bigtreetech would have to answer that one.

Reply:
All ini/variant stuff needs big revamp and unification imo, PeripheralPins is especially horrid mess. If we bump all STM32 to ststm32@14 (of higher) it's possible to merge variants, but i don't like too much defines left visibe for end user in ini... maybe move at least some to variant files with #ifdef?

Reply:
> It looks like we could define many of these, like `PIN_SERIAL1_RX`, in the `variant_*.h` file(s) (with `#ifndef`), and then only include the overrides in the env where they differ from typical defaults. Do board makers have wide discretion about these pins, or are these the serial ports that would always be defined based on the MCU?

Yes, all the -D define can be defined in `variant_*.h`, The reason why so many -D define are set in `.ini` is: `variant_*.h` and `variant_*.cpp` is directly copied from `framework-arduinoststm32\variants\STM32H7xx\H723Z(E-G)T_H730ZBT_H733ZGT` without modifying anything. 

Too many folders in `buildroot\share\PlatformIO\variants` are really redundant in my opinion. I hope there can be general libraries in `framework-arduinoststm32\variants\`. We only need to use -D define in `.ini` to adapt to different motherboards for the same product line MCU.

Of course, at present, `framework-arduinoststm32\variants\` is not directly universal, so my idea is redundant at the moment. We can directly move -D define in `.ini` to `variant_*.h`

Reply:
> Too many folders in buildroot\share\PlatformIO\variants are really redundant in my opinion.

The revision history gives clues about how we got here, basically by contributors using the output from the STMicro Cube Programmer and/or copying and modifying variants to gain full control over them. If it is possible to replace some or most of our custom variants with simple build flags, that would be very helpful to reduce some of our complexities. I'd like to see some examples of converting variants to build flags so that we have some idea about the procedure. It sounds like we would need to know details about the standard variants included with each version of the STSTM32 platform in order to be able to know what should or should not be changed via build flags.